.github/workflows/ci.yml

```diff
@@ -9,6 +9,24 @@ on:
             - main
 
 jobs:
+    type_check:
+        name: Type Check
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: "lts/*"
+
+            - name: Install Packages
+              run: |
+                  npm install
+                  npm run build
```

Comment:
The types definition files aren't there until we build.

Reply:
If that is so, doesn't package build step fail when the type is wrong for the package? 🤔 do we need another tsc command to run separately on the main repo?

Reply:
Yeah that is true. That's a pain I've just dealt with in other repos. In theory we could do a strategy like `|| 0` to ignore any failures.

Reply:
just to re iterate, my doubt was why we'd need to run `build` command and `test:types` on source files? 
- the build process itself has a in-built `tsc` check? if the build fails then `tsc` is also expected to fail? 
- we cannot have a `tsc` check separately because its dependent on `build` (which in itself has a `tsc` check?)

now the question resolves to do we have any other `ts` files apart from the `ts` files that are built in the repo? if yes then its better to have a `tsc` for fallback

(sorry for the confusion 😅)

Reply:
That's a good point. I hadn't noticed that `test:types` just runs `tsc`...for some reason I had it in my head that there was something else running there.

So yes, `npm run build` actually runs `tsc`, so we don't need something separate running `tsc`

Reply:
Just clarifying - is there a request for changes?

Reply:
Yes, there's no need to run `tsc` separately because that's already happening via `npm run build`.

Reply:
Ah got it. Thanks, I see now - sorry for giving you the runaround harish-sethuraman.